tests/integration/sync/test_sync_code.py

```diff
@@ -0,0 +1,207 @@
+import os
+import platform
+
+import logging
+import json
+import shutil
+import tempfile
+import time
+import uuid
+from pathlib import Path
+from unittest import skipIf
+
+import pytest
+import boto3
+
+from samcli.lib.utils.resources import (
+    AWS_APIGATEWAY_RESTAPI,
+    AWS_LAMBDA_FUNCTION,
+    AWS_STEPFUNCTIONS_STATEMACHINE,
+)
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+IS_WINDOWS = platform.system().lower() == "windows"
+# Some wait time for code updates to be reflected on each service
+LAMBDA_SLEEP = 3
+API_SLEEP = 5
+SFN_SLEEP = 5
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+
+LOG = logging.getLogger(__name__)
+
+TEMP_DIR = ""
+STACK_NAME = ""
+TEMPLATE_PATH = ""
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+class TestSyncCode(SyncIntegBase):
+    @pytest.fixture(autouse=True, scope="class")
+    def sync_code_base(self):
+        global TEMP_DIR
+        global STACK_NAME
+        global TEMPLATE_PATH
```

Comment:
All caps is also for constants and not global variables.

Reply:
No, those variables will not be shared between instances. Since each test is an object by there own, setting self.var = value in the fixture cannot be applied to all test instances (it's not like setUp() where it's called upon each test object instantiation.) So it needs to be static and global. I can try to make it static class variable like TestSyncCode.temp_dir and see if it works equivalently (I would expect so)

Reply:
You can use `request` object and then assign values to the `cls` variable. I've managed to do an example in draft `sam logs` PR: https://github.com/aws/aws-sam-cli/pull/3607/files#diff-c8df8b464cdea941c1cb9dbcdb4b335317ded559bf8835e966e482f7c5694bedR51